qubekit/fragmentation/WBOFragmenter.py

```diff
@@ -0,0 +1,228 @@
+            off_molecule, target_bond_smarts=self.rotatable_smirks
+        )
+
+        # remove duplicates
+        fragmentation = self._deduplicate_fragments(fragmentation)
+
+        # visualise results in a file
+        depict_fragmentation_result(result=fragmentation, output_file="fragments.html")
+
+        # convert the results to QUBEKit Ligand
+        qk_molecule = self._results_to_ligand(fragmentation)
+
+        return qk_molecule
+
+    def _deduplicate_fragments(
+        self, fragmentation_result: "FragmentationResult"
+    ) -> "FragmentationResult":
+        """Remove symmetry equivalent fragments from the results."""
+        unique_fragments = []
+
+        # group by the same smiles
+        for smile, fragments in itertools.groupby(
+            fragmentation_result.fragments,
+            lambda frag: frag.molecule.to_smiles(explicit_hydrogens=False),
+        ):
+            fragments = list(fragments)
+
+            if len(fragments) == 1:
+                unique_fragments.extend(fragments)
+                continue
+
+            symmetry_groups = set()
+            for fragment in fragments:
+                bond_map = fragment.bond_indices
+                fragment_mol = fragment.molecule
+                # get the index of the atoms in the fragment
+                atom1, atom2 = self._get_atom_index(
+                    fragment_mol, bond_map[0]
+                ), self._get_atom_index(fragment_mol, bond_map[1])
+                symmetry_classes = self._rd_get_atom_symmetries(fragment_mol)
+                symmetry_group = tuple(
+                    sorted([symmetry_classes[atom1], symmetry_classes[atom2]])
+                )
+                if symmetry_group not in symmetry_groups:
+                    symmetry_groups.add(symmetry_group)
+                    unique_fragments.append(fragment)
+
+        fragmentation_result.fragments = unique_fragments
+
+        return fragmentation_result
+
+    def _rd_get_atom_symmetries(self, molecule: "Molecule") -> List[int]:
+
+        from rdkit import Chem
+
+        rd_mol = molecule.to_rdkit()
+        return list(Chem.CanonicalRankAtoms(rd_mol, breakTies=False))
+
+    def _get_atom_index(self, molecule: "Molecule", map_index: int) -> int:
+        """Returns the atom index of the atom in a molecule which has the specified map
+        index.
+
+        Parameters
+        ----------
+        molecule
+            The molecule containing the atom.
+        map_index
+            The map index of the atom in the molecule.
+
+        Returns
+        -------
+            The corresponding atom index
+        """
+        inverse_atom_map = {
+            j: i for i, j in molecule.properties.get("atom_map", {}).items()
+        }
+
+        atom_index = inverse_atom_map.get(map_index, None)
+
+        assert (
+            atom_index is not None
+        ), f"{map_index} does not correspond to an atom in the molecule."
+
+        return atom_index
```

Comment:
This could be made to work with our internal atom maps now and could probably be quite useful as a method on `Ligand` some like `Ligand.get_atom_with_map_idx` then the numerical index is just `Atom. atom_index`.